apps/api/pages/api/destination-calendars/[id]/_patch.ts

```diff
@@ -56,16 +62,250 @@ import { schemaQueryIdParseInt } from "~/lib/validations/shared/queryIdTransform
  *       404:
  *        description: Destination calendar not found
  */
+type DestinationCalendarType = {
+  userId?: number | null;
+  eventTypeId?: number | null;
+  credentialId: number | null;
+};
+
+type UserCredentialType = {
+  id: number;
+  appId: string | null;
+  type: string;
+  userId: number | null;
+  user: {
+    email: string;
+  } | null;
+  teamId: number | null;
+  key: Prisma.JsonValue;
+  invalid: boolean | null;
+};
+
 export async function patchHandler(req: NextApiRequest) {
-  const { prisma, query, body } = req;
+  const { userId, isAdmin, prisma, query, body } = req;
   const { id } = schemaQueryIdParseInt.parse(query);
   const parsedBody = schemaDestinationCalendarEditBodyParams.parse(body);
+  const assignedUserId = isAdmin ? parsedBody.userId || userId : userId;
 
+  validateIntegrationInput(parsedBody);
+  const destinationCalendarObject: DestinationCalendarType = await getDestinationCalendar(id, prisma);
+  await validateRequestAndOwnership({ destinationCalendarObject, parsedBody, assignedUserId, prisma });
+  if (parsedBody.eventTypeId) parsedBody.userId = undefined;
```

Comment:
great refactor btw, I would change two things and we are good to merge, first here, I would try to find a better solution, reassigning objects values like that in the middle of the handler is prone to potential problems in the future.

as the function gets bigger for example, someone might forgot that this was done. 

also this is done in the middle of the handler without context as to why it is done.

what I would recommend if it is possible:

try to do this replacement where is makes sense in the code, like directly inside the impacted prisma queries for example